cgp/population.py

```diff
@@ -48,61 +53,67 @@ def __init__(
         self._generate_random_parent_population()
 
     @property
-    def champion(self) -> Individual:
+    def champion(self) -> IndividualBase:
         """Return parent with the highest fitness.
         """
         return max(self._parents, key=lambda ind: ind.fitness)
 
     @property
-    def parents(self) -> List[Individual]:
+    def parents(self) -> List[IndividualBase]:
         return self._parents
 
     @parents.setter
-    def parents(self, new_parents: List[Individual]) -> None:
+    def parents(self, new_parents: List[IndividualBase]) -> None:
         self.generation += 1
         self._parents = new_parents
 
-    def __getitem__(self, idx: int) -> Individual:
+    def __getitem__(self, idx: int) -> IndividualBase:
         return self._parents[idx]
 
     def _generate_random_parent_population(self) -> None:
         self._parents = self._generate_random_individuals(self.n_parents)
         self._label_new_individuals(self._parents)
 
-    def _label_new_individuals(self, individuals: List[Individual]) -> None:
+    def _label_new_individuals(self, individuals: List[IndividualBase]) -> None:
         for ind in individuals:
             ind.idx = self._max_idx + 1
             self._max_idx += 1
 
-    def _generate_random_individuals(self, n: int) -> List[Individual]:
+    def _generate_random_individuals(self, n: int) -> List[IndividualBase]:
         individuals = []
         for i in range(n):
             if isinstance(self._genome_params, dict):
-                individual = Individual(fitness=None, genome=None)
-            elif isinstance(self._genome_params, list) and isinstance(
-                self._genome_params[0], dict
-            ):
-                individual = IndividualMultiGenome(fitness=None, genome=None)
+                genome: Genome = Genome(**self._genome_params)
+                genome.randomize(self.rng)
+                individual_s = IndividualSingleGenome(
+                    fitness=None, genome=genome
+                )  # type: IndividualBase
```

Comment:
```suggestion
                )
```

Reply:
or what is this needed for?

Reply:
It is needed by mypy. I added an explanatory comment behind this.